benchmark/utils/general.hpp

```diff
@@ -329,6 +337,40 @@ void backup_results(rapidjson::Document& results)
 }
 
 
+inline std::shared_ptr<gko::CudaAllocatorBase> create_cuda_allocator()
+{
+    std::string flag{FLAGS_allocator};
+    if (flag == "default") {
+        return std::make_shared<gko::CudaAllocator>();
+    } else if (flag == "async") {
```

Comment:
Is there any point in having an async allocator if you are not using a stream ?

Reply:
The default stream is also a stream, the stream-ordered semantics of the allocator still apply

Reply:
Does stream==nullptr mean that it is the default stream ? I cannot seem to find it in the docs anywhere.

Reply:
It's listed as 0 in there

Reply:
but the default stream does not allow other streams created by `cudaStreamNonBlocking` execute at the same time, right?

Reply:
yes, but the benchmarks only use a single stream, so we are fine for now. As soon as we move to multi-stream setups, we need to use a custom stream or `CU_STREAM_PER_THREAD`